x/config/builder.go

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package config
+
+import (
+	"fmt"
+	"strings"
+
+	"go.uber.org/yarpc"
+	"go.uber.org/yarpc/api/transport"
+)
+
+// Builder TODO
+type Builder struct {
+	Name       string
+	Inbounds   []InboundConfig
+	Outbounds  []OutboundConfig
+	Transports []TransportConfig
```

Comment:
I would like to figure out a place to put procedures too, yes. We can't really read them from the configuration file.

One way would be to provide `Register([]transport.Procedure)` functions on the Configurator but that introduces a circular dependency: Handlers depend on the Dispatcher existing upon construction (so that they can grab clients for the service calls they need to make), and with this change the Dispatcher will depend on Handlers existing beforehand. Maybe we use a tchannel-go like approach of accepting a RegisterFunc?

    type RegisterFunc func(Dispatcher)

Ideally I would like whatever we pick here to play nicely with `dig`, injecting the resources required to build the handlers and giving us procedures back that we can register on the dispatcher.